src/main/scala/rocket/Events.scala

```diff
@@ -28,14 +28,15 @@ class EventSet(gate: (UInt, UInt) => Bool, val events: Seq[(String, () => Bool)]
 class EventSets(val eventSets: Seq[EventSet]) {
   def maskEventSelector(eventSel: UInt): UInt = {
     // allow full associativity between counters and event sets (for now?)
-    val setMask = (BigInt(1) << log2Ceil(eventSets.size)) - 1
-    val maskMask = ((BigInt(1) << eventSets.map(_.size).max) - 1) << eventSetIdBits
+    val setMask = (BigInt(1) << eventSetIdBits) - 1
+    val maskMask = ((BigInt(1) << eventSets.map(_.size).max) - 1) << maxEventSetIdBits
     eventSel & (setMask | maskMask).U
   }
 
   private def decode(counter: UInt): (UInt, UInt) = {
-    require(eventSets.size <= (1 << eventSetIdBits))
-    (counter(log2Ceil(eventSets.size)-1, 0), counter >> eventSetIdBits)
+    require(eventSets.size <= (1 << maxEventSetIdBits))
+    require(eventSetIdBits > 0)
+    (counter(eventSetIdBits-1, 0), counter >> maxEventSetIdBits)
```

Comment:
on the high level, what is the purpose of this change? 
differentiate the actually eventSet and maxEventSet? and maxEventSet will be limited by the architecture?

Reply:
just making the code the same for both `EventSet` and `SuperscalarEventSet`